contrib/utilities/indent_common.sh

```diff
@@ -0,0 +1,300 @@
+#!/bin/bash
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2018 - 2021 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This file contains a number of common functions used all indent scripts
+#
+
+#
+# This function checks that we are in the root directory and that shfmt is
+# available. It is called by both indent.sh and indent_all.sh to ensure that
+# the rest of the indentation pipeline makes sense.
+#
+
+#
+# DEAL_II_SHFMT can be used to change the default version of shfmt
+#
+
+export DEAL_II_SHFMT="${DEAL_II_SHFMT:-shfmt}"
+export DEAL_II_SHFMT_OPTIONS="-ln bash -i 2 -bn -ci -sr"
+
+checks() {
+  if test ! -d contrib -o ! -d deal.II-toolchain; then
+    echo "*** This script must be run from the top-level directory of candi."
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v "${DEAL_II_SHFMT}")" ]; then
+    echo "***   No shfmt program found."
+    echo "***"
+    echo "***   Install shfmt via snap: sudo snap install shfmt"
+    exit 1
+  fi
+
+  # Make sure to have the right version.
+  SHFMT_VERSION="$(${DEAL_II_SHFMT} --version)"
+  SHFMT_MAJOR_VERSION=$(echo "${SHFMT_VERSION}" | sed 's/^[^0-9]*\([0-9]*\).*$/\1/g')
+  SHFMT_MINOR_VERSION=$(echo "${SHFMT_VERSION}" | sed 's/^[^0-9]*[0-9]*\.\([0-9]*\).*$/\1/g')
+
+  if [ "${SHFMT_MAJOR_VERSION}" -ne 3 ] || [ "${SHFMT_MINOR_VERSION}" -ne 3 ]; then
+    echo "***   This indent script requires shfmt version 3.3.1,"
+    echo "***   but version ${SHFMT_MAJOR_VERSION}.${SHFMT_MINOR_VERSION} was found instead."
+    echo "***"
+    echo "***   Install shfmt via snap: sudo snap install shfmt"
```

Comment:
This won't give you shfmt 3.3.1 in the future. Is this strict requirement necessary? (does the formatting change?)

Reply:
I adapted the version check to test for the current stable version 3.3.1 or later. The formatting is set through the options passed to shfmt. Don't know if they would change at any future time.